### PR TITLE
Sample schema refactor

### PR DESCRIPTION
**Pull request by sveinugu**
Refactored sample schema to allow for several variants of sample_types, dependent on the biospecimen_class property (renamed from biomaterial_type). Added ontology validation of biospecimen_class terms, and ancestor validation to all new properties.

### CODE REVIEW

**Comment by sveinugu** [comment]
Renamed property:

  - biomaterial_type -> biospecimen_class.
    - Now uses ontology terms from NCI Thesaurus, limited to four terms: Cell, Abnormal Cell, Cell Line, Organism Part. 
    - Previous "primary cell culture" has been merged with "primary cell", now "Cell".

New properties:
  - cell_type
  - abnormal_cell_type (typically cancer cells)
  - cell_line
  - organism_part (typically tissue/organ)
  - main_type
    - autogenerated from the other ones, to have a single summary attribute

Validation
  - Added in-schema validation of dependency between biospecimen_class and the other types, using the new 'requireIf" attribute in the OPML files (mechanism committed in master, now used here).
  - Added "ancestry" attribute (mechanism committed in master, now used here)

**Comment by sveinugu** [comment]
Fixes: https://github.com/fairtracks/fairtracks_standard/issues/15

**Comment by sveinugu** [comment]
Referenced in https://github.com/orgs/fairtracks/teams/fairtracks-developers/discussions/20

**Comment by jmfernandez** [comment]
Hi @sveinugu , some of the functionality you are looking for is already implemented, with the key name of `ancestors`, as you can read in the documentation:

https://github.com/fairtracks/fairtracks_validator/blob/master/README.md

Could you open issues in [FAIRTracks Validator repo](https://github.com/fairtracks/fairtracks_validator/issues) describing the new features you are missing, please?

File: json/overview/fairtracks_sample.overview.opml [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
     <outline _text="biospecimen_class" description="Main type of structural unit to be used for classification of biospecimen contents" required="true" type="object">
-      <outline _text="term_url" description="URL linking to an NCIT ontology term" enum="http://purl.obolibrary.org/obo/NCIT_C12508|http://purl.obolibrary.org/obo/NCIT_C12801|http://purl.obolibrary.org/obo/NCIT_C16403|http://purl.obolibrary.org/obo/NCIT_C103199 " examples="http://purl.obolibrary.org/obo/NCIT_C16403" format="term" matchType="exact" ontology="http://purl.obolibrary.org/obo/ncit/releases/2019-09-09/ncit.owl" required="true" type="string" />
+      <outline _text="term_url" description="URL linking to an NCIT ontology term" enum="http://purl.obolibrary.org/obo/NCIT_C12508|http://purl.obolibrary.org/obo/NCIT_C12913|http://purl.obolibrary.org/obo/NCIT_C16403|http://purl.obolibrary.org/obo/NCIT_C103199 " examples="http://purl.obolibrary.org/obo/NCIT_C16403" format="term" matchType="exact" ontology="http://purl.obolibrary.org/obo/ncit/releases/2019-09-09/ncit.owl" required="true" type="string" />
```
Comment (jmfernandez) [inline json/overview/fairtracks_sample.overview.opml]: @sveinugu There is a whitespace at the end of the enum declaration

**Comment by dzerbino** [comment]
Looks good to me.

**Comment by sveinugu** [comment]
Force pushed a rebased and fixed commit

**Pull request merged**